### PR TITLE
support cache 272 dentries at least and some misc optimizations

### PR DESCRIPTION
**Pull request by YuezhangMo**

### CODE REVIEW

**Comment by namjaejeon** [comment]
@hclee Please review this PR. Thanks!

**Comment by namjaejeon** [comment]
@YuezhangMo Can you send this PR again to run auto-test by github action ?

**Comment by hclee** [comment]
> @hclee Please review this PR. Thanks!

Okay, I will on Monday.

File: lib/exfat_fs.c [hunk 1]
```diff
@@ -160,24 +161,34 @@ struct exfat *exfat_alloc_exfat(struct exfat_blk_dev *blk_dev, struct pbr *bs)
 		goto err;
 	}
 
+	/*
+	 * A dentry set can have up to 256 dentries, occupying 8KB.
+	 * To keep the entire dentry set in buffers and ->buffer_count
+	 * to be a power of 2, set the total buffers size to 16KB.
+	 */
+	exfat->buffer_count = MAX(16 * KB / exfat->clus_size,
```
Comment (hclee) [inline lib/exfat_fs.c:169]: Can you elaborate why this is needed?
Other than this, Looks good to me.

**Comment by hclee** [comment]
Ah, I mean 16*KB / exfat->clus_size seems unnecessary to calculate
buffer_count.
I asked a question ambigously.

2024년 1월 15일 (월) 오후 7:54, Yuezhang Mo ***@***.***>님이 작성:

**Comment by YuezhangMo** [comment]
> Ah, I mean 16*KB / exfat->clus_size seems unnecessary to calculate buffer_count.

I changed to use `((MAX_EXT_DENTRIES + 1) * DENTRY_SIZE) / exfat_get_read_size(exfat) + 1` to calculate buffer_count.

**Comment by hclee** [comment]
Looks good to me.

**Comment by namjaejeon** [comment]
@YuezhangMo It would be grateful if you run checkpatch.pl before sending a PR.

```
0002-lib-remove-zero_cluster-from-struct-exfat.patch
----------------------------------------------------
WARNING: Missing a blank line after declarations
#106: FILE: lib/libexfat.c:229:
+		int iter_size = MIN(size, sizeof(zero_buf));
+		if (iter_size != write(fd, zero_buf, iter_size))

0007-lib-support-512-bytes-cluster-size.patch
---------------------------------------------
WARNING: calloc uses number as first arg, sizeof is generally wrong
#308: FILE: lib/exfat_fs.c:179:
+	bd = (struct buffer_desc *)calloc(sizeof(*bd), exfat->buffer_count);

```

**Comment by YuezhangMo** [comment]
Hi @namjaejeon 

> It would be grateful if you run checkpatch.pl before sending a PR.

OK, I will run checkpatch.pl next time.

Is it possible to run checkpatch.pl in CI?

> WARNING: calloc uses number as first arg, sizeof is generally wrong

This issue existed before this PR, I created 45a34930 to fix it.

I found there are some unnecessary casts for the returns of calloc()/malloc(), I created 5192f766 to remove it.

**Comment by namjaejeon** [comment]
> Is it possible to run checkpatch.pl in CI?

Yes. possible. need to download kernel source for this.

Okay, I have applied all patches. thanks for your work!

**Comment by namjaejeon** [comment]
@YuezhangMo I have forwarded szbot report mail to you. Your valid size patches seem to cause this. We need to fix a kernel oops as soon as possible.

**Pull request closed without merging**